drivers/modem/Kconfig.gsm

```diff
@@ -36,6 +36,24 @@ config MODEM_GSM_QUECTEL
 
 endchoice
 
+choice MODEM_GSM_STATUS_COMMAND_TYPE
+	prompt "Select AT command Type"
+	default MODEM_GSM_SELECT_CREG
+	help
+	  Use particular type of AT command for registration status.
+
+config MODEM_GSM_SELECT_CREG
+	bool "CREG command"
```

Comment:
I think this can be a string type with `"CREG"` as the default value

Reply:
Thanks. I tried but zephyr is not allowing me to use string as a type of choice. i do not find any example either. is there any?

Reply:
oops, didn't know that, how about 
```kconfig
config MODEM_GSM_STATUS_COMMAND
  string "Status command"
  default "CREG" if MODEM_GSM_SELECT_CREG
  default "CEREG" if MODEM_GSM_SELECT_CREG
```
?

I found some similar things here: `zephyrproject/modules/hal/espressif/components/esptool_py/Kconfig.projbuild`

In any case, you should really create another PR against `main`